src/CloudinaryAdapter.php

```diff
@@ -283,7 +283,7 @@ public function getTimestamp($path)
         return $this->getMetadata($path);
     }
 
-    private function normalizeMetadata($resource)
+    protected function normalizeMetadata($resource)
```

Comment:
is it really necessary to be protected? Actually, I cannot imagine a situation when John Doe needs to extend the adapter and redefine this function.

Reply:
Hi @enl 
Well, actually I needed it. But of course there is still composition as an alternative to inheritance, but it's always better if libraries give you the freedom to decide.

But the other fix is more important

Reply:
I mean every `protected` is actually API we need to stick to, since easily can be redefined, so every change in its declaration can break backward compatibility. That's why I am usually very careful in exposing implementation details.

Can you please provide me an example when you need it?

Reply:
We make a lot use of directory structures in Cloudinary, e.g. we have different `CloudinaryVolumes` which point do different folders (and have subfolders).

To separate the different volumes pointing to the same account, we added a `basePath`, which will be automatically prepended before each path. But this means we have to remove it again in the returned file list to receive the format of the original path. So it's something like an `CustomCloudinaryAdapter`

Reply:
So, as far as I understand you need some conversion logic from $path to $publicId and opposite? 

As for me, it will be nicer to introduce it as a some kind of converter. BTW, this feature will be necessary in #16 (since `FilesystemInterface` does not allow to `getAdapter`, I will need to inject api facade into plugin and reuse this path-to-id conversion there (or just move it into api level)).

So, it will be nice, if you return private _here_, and then create an issue describing your case and we definitely work out proper solution :)

Reply:
Hi @swey! Finally I have enough time to continue moving this project to 1.1 release. I've just closed #19 issue about PathConverter. Now you can implement this interface to achieve desired behavior in nicer and more OO-like way :)